keys/keys.go

```diff
@@ -355,3 +355,14 @@ func MetaScanBounds(key proto.Key) (proto.Key, proto.Key) {
 	// Otherwise find the first entry greater than the given key in the same meta prefix.
 	return key.Next(), proto.Key(key[:len(Meta1Prefix)]).PrefixEnd()
 }
+
+// MetaReverseScanBounds returns the start and end keys of the range within which the
+// desired meta record can be found by means of an engine reverse scan. The given key
+// must be a valid RangeMetaKey as defined by ValidateRangeMetaKey.
+func MetaReverseScanBounds(key proto.Key) (proto.Key, proto.Key, error) {
```

Comment:
Should add some simple test cases for this function similar to `TestMetaScanBounds`.